test/Table_Tests/src/Formatting/Parse_Values_Spec.enso

```diff
@@ -626,6 +626,19 @@ add_specs suite_builder =
             r8.to_vector . should_equal [Nothing, Nothing, Nothing]
             Problems.assume_no_problems r8
 
+        group_builder.specify "should be able to parse numeric date formats with a Data_Formatter" <|
+            formatter = Data_Formatter.Value . with_datetime_formats date_formats=["yyyyMMdd"]
+
+            c1 = Column.from_vector "A" ["20241018", "20240315", "20240917"]
+            r1 = c1.parse format=formatter
+            r1.value_type.should_equal Value_Type.Date
+            r1.to_vector.should_equal [Date.new 2024 10 18, Date.new 2024 3 15, Date.new 2024 9 17]
+
+            c2 = Column.from_vector "A" ["20241018", "2024", "20240917"]
+            r2 = c2.parse format=formatter
+            r2.value_type.should_equal Value_Type.Integer
+            r2.to_vector.should_equal [20241018, 2024, 20240917]
```

Comment:
I'm slightly anxious about this change as it feels like unless specified as a date, `202041104` is first a number and date second.

BUT I assume this only works if a format like `yyyyMMdd` is added to 'expected formats'. So I think it is good as long as we don't make this format one of the defaults.

I.e. IMHO (open to being convinced otherwise), in a vanilla parser setting strings like `202041104` should be treated as numbers (which seems to be the case currently 👍). Only if the user _opts-in_ by adding a date format like `yyyyMMdd`, then we change the precedence and valid eight digit numbers become dates. Does that make sense?

Reply:
It is a number unless you have provided a formatter with an explicit `yyyyMMdd` date format.
Note `c1.parse format=formatter`. 
I'll add a test to show `c1.parse` is still an integer.

Reply:
> Added a test for default parse to ensure not being parsed incorrectly without the opt in.

Great, that's a very good idea